### ISSUE TITLE
Can't enter some Groups & soft crash

### ISSUE THREAD

**Issue by Beryesa** [labnex/LabNex#16]
**I confirm that I have searched the issue tracker**
Before opening an issue, take a few minutes to search the issue tracker (both open and closed issues) for the problem you are facing. This will reduce the need for issue triage and save time.

 - [x] I confirm that I have searched the issue tracker

**Describe the bug**

Groups page crashes on some main groups, works fine on its subgroups though.

**To Reproduce**

Try exploring your groups
Throws you back 
Try multiple times
"Application keeps stopping" triggers the log report page (which doesn't work on Fdroid?)

**Expected behavior**
It can browse all the groups

**Logs**

![Log_2024-10-10_06-17-25.txt](https://github.com/user-attachments/files/17322142/Log_2024-10-10_06-17-25.txt)

**Versions (please complete the following information)**
 - LabNex:  1.0.1
 - Android:  14

**Mode of installation**
 - [ ] Google Play Store
 - [x] F-droid
 - [ ] APK
 - [ ] Built from source

Thanks for the app!

**Comment by licaon-kter** [comment]
Was your token set for all the permissions?

**Comment by Beryesa** [comment]
> Was your token set for all the permissions?

I had given api, read_api and read_user

Now I tried again with read_/write_repository (though LabNex doesn't support/need http anyway, why would it matter) added as well, but the same issue persists.
I can see some groups but not all, I can almost say only the ones with more nested subgroups but I don't think it's that obvious.

**Comment by licaon-kter** [comment]
@Beryesa read https://github.com/labnex/LabNex/issues/14#issuecomment-2395549134 and try with a token that covers all those too

**Comment by mmarif4u** [comment]
@Beryesa thanks for sharing the log, I will look into it.

As mentioned by @licaon-kter try with token with most API scopes selected.

> Now I tried again with read_/write_repository (though LabNex doesn't support/need http anyway, why would it matter) added as well, but the same issue persists.

The common concept here is that most people think of repository as a repository, while in GitLab it is the Files API. So not giving this permission you will be not able to browse files or will have crashes with the app.
In GitLab repository is called project, organization is called group and repository is called for Files (at least through API). But I am sure most of us know that. That's why the app sign in screen ask if not all most API scopes.

Other than that, can you detail out if the groups you are accessing belongs to you, or you are part of it?

**Comment by mmarif4u** [comment]
After checking the logs, the crash happens when you are visiting a group. Somehow it crashed when fetching the group projects.

Please test this APK and let me know if the crash still occurs. Remove F-droid version first.

[LabNex-debug.zip](https://github.com/user-attachments/files/17329611/LabNex-debug.zip)

**Comment by Beryesa** [comment]
Now it doesn't crash but doesn't show the subgroups either (on those that previously crashed it)
Yet some groups work (they also did before)
What might be the difference?

**Comment by mmarif4u** [comment]
Currently, there are no subgroups implemented, so if you are seeing subgroups in the app, I’d be curious to know where they are coming from.

When you tap on the Groups menu on the main screen, it takes you to the groups screen (no subgroups, only top-level groups). After tapping on a group, you will see its details and the projects under it.

I recommend creating a new issue with a feature request for subgroups.

**Comment by Beryesa** [comment]
Let me Rephrase,
In the Groups page I see:
Main Groups
Groups under those MG (Subgroups)
![Screenshot_2024-10-10-19-22-18-351_com.labnex.app-edit.jpg](https://github.com/user-attachments/assets/63dfe5cd-e767-44f7-a1e0-2a743998fe97)

That works already, somehow 

Projects are shown under those Subgroups
![Screenshot_2024-10-10-19-29-05-203_com.labnex.app-edit.jpg](https://github.com/user-attachments/assets/bbb6a0a4-d1e5-4b08-a57c-62183b22b39d)


But Subgroups and projects are not shown in the Main Groups
![Screenshot_2024-10-10-19-27-53-075_com.labnex.app-edit.jpg](https://github.com/user-attachments/assets/ee907ca1-468c-43a3-9be2-f18155f66fa7)

I should probably make another issue as advised though, it doesn't crash anymore, thanks!

**Comment by mmarif4u** [comment]
Thanks for the details. To be honest, I wasn’t expecting subgroups with the main groups API. I’m not sure how GitLab handles this, but I’ll leave that to them.

Regarding our discussion, as I mentioned above, subgroups under the main group have not been implemented yet. I see you’ve opened a new issue for it—thank you.
